hackathon_site/event/api_views.py

```diff
@@ -66,3 +67,41 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+    lookup_field = "team_code"
+    queryset = Team
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
+
+        profile = request.user.profile
+        current_team = profile.team
+
+        self.kwargs["lookup_url_kwarg"] = team_code
```

Comment:
`lookup_url_kwarg` is the URL keyword argument that gets used in filtering. It basically does:

```python
team = EventTeam.objects.filter(**{self.lookup_field: self.kwargs[self.lookup_url_kwarg]})
```

Which we want to be

```python
EventTeam.objects.filter(team_code=team_code)
```

`self.lookup_url_kwarg` will default to `self.lookup_field`, so you don't need to do this. This is setting it on self.kwargs anyway, which is doesn't do anything.